src/Microsoft.AspNet.SignalR.Client/Connection.cs

```diff
@@ -137,10 +137,11 @@ public class Connection : IConnection, IDisposable
         /// </summary>
         /// <param name="url">The url to connect to.</param>
         public Connection(string url)
-            : this(url, (string)null)
+            :this(StripQueryString(url), GetQueryStringFromUrl(url))
```

Comment:
One thing I did not notice before - the splitting should happen in https://github.com/SignalR/SignalR/blob/1fba14fa3437e24c204dfaf8a18db3fce8acad3c/src/Microsoft.AspNet.SignalR.Crank/DefaultConnectionFactory.cs#L17. Doing this in Connection.cs changes the current behavior which I don't want to avoid (I don't know how big/small it is but we should not change it)

Actuall now I see - before we would throw if url contained `?` 
https://github.com/SignalR/SignalR/blob/dev/src/Microsoft.AspNet.SignalR.Client/Connection.cs#L166 and after this change we wouldn't but `Connection` should not be changed for this class.

Reply:
Yes that makes much more sense, I hadn't considered making the change there since we'd reverted the query string crank argument addition. I'll work on these changes tomorrow. Thanks